src/Checkers/ShapeTypeChecker.php

```diff
@@ -0,0 +1,48 @@
+<?php
+namespace Prezly\PropTypes\Checkers;
+
+use Prezly\PropTypes\Exceptions\PropTypeException;
+
+class ShapeTypeChecker implements TypeChecker
+{
+    /** @var \Prezly\PropTypes\Checkers\TypeChecker[] */
+    private $shape;
```

Comment:
I've been trying to follow the original implementation (including var names) as close as it's still practical.

In `facebook/prop-types` this parameter is called **shape types** (for a good reason ;)). Thus I'd recommend to use the same naming here as well. `$shape_types` it is.

Would you agree?